In [1]:
import numpy as np
import pandas as pd
from scipy import stats

## Задача 1

_Даны значения величины заработной платы заемщиков банка (zp) и значения их поведенческого кредитного скоринга (ks):_  
> zp = [35, 45, 190, 200, 40, 70, 54, 150, 120, 110]  
> ks = [401, 574, 874, 919, 459, 739, 653, 902, 746, 832]  
>
_Найдите ковариацию этих двух величин с помощью элементарных действий, а затем с помощью функции cov из numpy. Полученные значения должны быть равны._  
_Найдите коэффициент корреляции Пирсона с помощью ковариации и средних квадратических отклонений двух признаков, а затем с использованием  функций из библиотек numpy и pandas._

1. Найдем ковариацию этих двух величин  
    
1.1 Без использования специальных функций


In [19]:
zp  = np.array([35, 45, 190, 200, 40, 70, 54, 150, 120, 110])
ks = np.array([401, 574, 874, 919, 459, 739, 653, 902, 746, 832])

cov_r = (zp * ks).mean() - zp.mean() * ks.mean()
print("Ковариация: ", cov_r)



Ковариация:  9157.839999999997


    
1.2 С помощью функции numpy.cov

In [20]:
cov = np.cov(zp, ks, ddof=0)
print("Ковариация: ", cov[1, 0])


Ковариация:  9157.84


2. Найдем коэффициент корреляции

2.1 Без использования специальных функций

In [7]:
corr = cov_r / (zp.std() * ks.std())
print("Коэффициент корреляции: ", corr)

Коэффициент корреляции:  0.8874900920739158


2.2 с помощью специальных функций из библиотек numpy и pandas

In [9]:
df = pd.DataFrame({'ZP': zp, 'KS': ks})
df[['ZP', 'KS']].corr()


ZP       KS
ZP  1.00000  0.88749
KS  0.88749  1.00000

In [8]:
corr1 = np.corrcoef(zp, ks)[1, 0]
print("Коэффициент корреляции: ", corr1)

Коэффициент корреляции:  0.8874900920739162


## Задача 2

_Измерены значения IQ выборки студентов, обучающихся в местных технических вузах:_  
> 131, 125, 115, 122, 131, 115, 107, 99, 125, 111  

_Известно, что в генеральной совокупности IQ распределен нормально. Найдите_
_доверительный интервал для математического ожидания с надежностью 0.95_

In [12]:
iq = np.array([131, 125, 115, 122, 131, 115, 107, 99, 125, 111])

In [13]:
n = iq.shape[0]
m = iq.mean()
s = iq.std(ddof=1)
t = stats.t.ppf(0.975, 9)

Доверительный интервал для оценки истинного (генерального) значения:

In [14]:
print(m - t * s / np.sqrt(n),  m + t * s / np.sqrt(n))
stats.t.interval(0.95 , df=len(iq)-1, loc=np.mean(iq), scale=stats.sem(iq))

110.55608365158724 125.64391634841274


(110.55608365158724, 125.64391634841274)

Доверительный интервал:  $(110.556; 125.644)$

## Задача 3.

_Известно, что рост футболистов в сборной распределен нормально с  дисперсией генеральной совокупности, равной 25 кв.см. На выборки объёма 27, среднее выборочное составляет 174.2._  
>
_Найдите доверительный интервал для математического ожидания с надежностью 0.95._


$\underline{X}\pm Z_\frac{\alpha}{2}\cdot\frac{\sigma}{\sqrt{n}}$

$Z_\frac{\alpha}{2}=1.96$

$174.2\pm 1.96\cdot\frac{5}{\sqrt{27}}$


In [15]:
n = 27
m = 174.2
s = 5
z = 1.96
print(m - z * s / np.sqrt(n), m + z * s / np.sqrt(n))

172.31398912064722 176.08601087935276


Доверительный интервал: $[172.314;176.086]$